applications/dashboard/controllers/class.homecontroller.php

```diff
@@ -97,6 +99,15 @@ public function fileNotFound() {
     }
 
     /**
+     * Clears the url from the user's navigation preferences.
+     */
+    private function clearNavigationPreferences() {
+        $userModel = new UserModel();
+        $uri = Gdn::request()->getRequestArguments('server')['REQUEST_URI'];
```

Comment:
Same, that's `dashboard/home/filenotfound` here (or whichever endpoint we're redirected to)